### ISSUE TITLE
`SFTTrainer.train(resume_from_checkpoint=...)` is accepted and ignored

### ISSUE THREAD

**Issue by CodersAcademy006** [ellanorai/ThinkRL#103]
`SFTTrainer.train` accepts `resume_from_checkpoint`, documents it, and never reads it. `thinkrl/training/sft_trainer.py:321`:

```python
def train(self, resume_from_checkpoint: str | None = None) -> dict[str, Any]:
    """
    Run training.

    Args:
        resume_from_checkpoint: Path to checkpoint to resume from
    ...
```

The identifier appears exactly twice in the file, on the signature line and in the docstring. It appears nowhere in the method body, so no checkpoint is loaded, no optimizer state is restored and `global_step` starts at zero.

The failure is silent and expensive in a specific way: the run starts, the loss curve looks plausible because it is a fresh curve, and the operator only discovers the restart when the step count or the wall-clock estimate does not match. On a multi-day job that is a day lost.

`cli/main.py:66` exposes the same idea as `--resume/-r` on the `train` command, but that command body is a `# TODO: Implement actual training loop` stub at `:115`, so the flag is inert there too.

**Suggested fix**

Either implement it, loading model, optimizer, scheduler and `global_step` and skipping the consumed portion of the dataloader, or raise `NotImplementedError` when a non-`None` value is passed. Accepting a path and ignoring it is the one behaviour that cannot be detected from the outside.

I am happy to open a PR for this.
